bots/tests-scan

```diff
@@ -358,22 +339,69 @@ def cockpit_tasks(api, update, branch_contexts, repo, pull_data, pull_number, sh
 
             return False
 
-        if trigger_externals():
+        def get_externals():
+            result = []
             for proj_repo in testmap.projects():
                 if proj_repo == "cockpit-project/cockpit":
                     continue
                 for context in testmap.tests_for_project(proj_repo).get("master", []):
-                    repo_context = context + "@" + proj_repo
-
-                    status = statuses.get(repo_context, {})
-                    if login not in whitelist and status.get("description", github.NO_TESTING) == github.NO_TESTING:
-                        priority = github.NO_TESTING
-                        changes = { "description": github.NO_TESTING, "context": repo_context, "state": "pending" }
-                    else:
-                        (priority, changes) = prioritize(status, title, labels, baseline, repo_context, number)
-                    if not update or update_status(api, revision, repo_context, status, changes):
-                        # don't specify a ref to check out; make-source already checks out project master
-                        results.append((priority, "pull-%d" % number, number, revision, "", context, base, proj_repo, "pull/%s/head" % number))
+                    result.append(context + "@" + proj_repo)
+            return result
+
+        def is_valid_context(context):
+            return context.split("@")[0] in contexts
+
+        # Create list of statuses to process
+        todos = {}
+        for status in statuses: # Firstly add all valid contexts that already exist in github
+            if is_valid_context(status):
+                todos[status] = statuses[status]
+        if not statuses: # If none defined in github add basic set of contexts
+            for context in branch_contexts.get(base, contexts):
+                todos[context] = {}
+        if trigger_externals():
+            for context in get_externals():
+                if context not in todos:
+                    todos[context] = {}
+
+        for context in todos:
+            # Get correct project and branch. Ones from test name have priority
+            project = repo
+            branch = base
+            (os_scenario, _, override) = context.partition("@")
+            override = override.split("/")
+            if len(override) == 2:
+                project = "/".join(override)
```

Comment:
This should set `branch` to `master`. An external repo has no relation to this `project` and `base`, and giving a SHA for e. g. a cockpit PR will fail on e. g. a cockpit-podman checkout.

If it makes the transition easier, we can keep the "default to master if not given", but it'd IMHO be clearer to always specify the branch in the context if `@` is being used at all.